File: LeetCode/cpp/2124_check_if_all_as_appears_before_all_bs.cc
/*
Difficulty: Easy
Runtime: 0 ms (100%)
*/

#include <string>

class Solution {
 public:
  /// Time: O(n), Space: O(1)
  bool checkString(const std::string s) {
    bool foundB = false;
    for (const char ch : s) {
      if (ch == 'b') foundB = true;
      if (foundB && ch == 'a') return false;
    }

    return true;
  }
};

File: LeetCode/rust/2124_check_if_all_as_appears_before_all_bs.rs
/*
Difficulty: Easy
Runtime: 0 ms (100%)
*/

impl Solution {
    /// Time: O(n), Space: O(1)
    /// Would rather take s as &str.
    pub fn check_string(s: String) -> bool {
        let mut found_b = false;
        for &ch in s.as_bytes() {
            if ch == b'b' {
                found_b = true;
            }
            if found_b && ch == b'a' {
                return false;
            }
        }

        true
    }
}
